src/gui/MainWindow.cpp

```diff
@@ -1611,5 +1611,36 @@ void MainWindow::onProjectFileNameChanged()
 	this->resetWindowTitle();
 }
 
+CustomQMdiArea::CustomQMdiArea(QWidget* parent) :
+	QMdiArea(parent),
+	m_isBeingMoved(false)
+{}
+
+void CustomQMdiArea::mousePressEvent(QMouseEvent* event)
+{
+	m_lastX = event->x();
+	m_lastY = event->y();
+	m_isBeingMoved = true;
+}
+
+void CustomQMdiArea::mouseMoveEvent(QMouseEvent* event)
+{
+	if (m_isBeingMoved == false) { return; }
+	if (m_lastX != event->x())
+	{
+		horizontalScrollBar()->setValue(horizontalScrollBar()->value() + m_lastX - event->x());
+		m_lastX = event->x();
+	}
+	if (m_lastY != event->y())
+	{
+		verticalScrollBar()->setValue(verticalScrollBar()->value() + m_lastY - event->y());
+		m_lastY = event->y();
+	}
```

Comment:
We can add some whitespace for readability.
```suggestion
	if (m_isBeingMoved == false) { return; }

	if (m_lastX != event->x())
	{
		horizontalScrollBar()->setValue(horizontalScrollBar()->value() + m_lastX - event->x());
		m_lastX = event->x();
	}

	if (m_lastY != event->y())
	{
		verticalScrollBar()->setValue(verticalScrollBar()->value() + m_lastY - event->y());
		m_lastY = event->y();
	}
```